src/main/java/com/synopsys/integration/detect/lifecycle/boot/DetectBoot.java

```diff
@@ -134,7 +134,8 @@ public DetectBoot(DetectBootFactory detectBootFactory) {
         this.detectBootFactory = detectBootFactory;
     }
 
-    public DetectBootResult boot(DetectRun detectRun, String[] sourceArgs, ConfigurableEnvironment environment, EventSystem eventSystem, DetectContext detectContext) throws DetectUserFriendlyException, IOException, IllegalAccessException {
+    public Optional<DetectBootResult> boot(DetectRun detectRun, String[] sourceArgs, ConfigurableEnvironment environment, EventSystem eventSystem, DetectContext detectContext)
```

Comment:
Is there a case where we return an empty optional?

Reply:
Yes this code on line 184.

```
Optional<DetectBootResult> configurationResult = printConfiguration(printFull, detectConfiguration, eventSystem, detectInfo);
        if (configurationResult.isPresent()) {
            return configurationResult;
        }
```